sql-cli/sql_cli/workflow_directory_parser.py

```diff
@@ -0,0 +1,242 @@
+        Any relevant metadata from the file frontmatter will be passed to the TransformOperator,
+        though we do not pass parameter dependencies at this stage.
+
+        :return: a TransformOperator
+        """
+        kwargs = {
+            "conn_id": self.metadata.get("conn_id"),
+            "parameters": None,
+            "handler": None,
+            "database": self.metadata.get("database"),
+            "schema": self.metadata.get("schema"),
+            "python_callable": lambda: (str(self.path), None),
+            "sql": self.content,
+        }
+        if airflow.__version__.startswith("2.2."):
+            kwargs["op_args"] = []
+        return TransformOperator(**kwargs)
+
+
+class YamlFile(WorkflowFile):
+    file_type = YAML_FILE_TYPE
+    operator_instance_builder_callable_map = {LOAD_FILE_OPERATOR: get_load_file_instance}
+
+    def __init__(self, root_directory: Path, path: Path, target_directory: Path) -> None:
+        super().__init__(root_directory, path, target_directory, YamlFile.file_type)
+        with open(path) as yaml_file:
+            self.yaml_content = yaml.load(yaml_file, Loader=yaml.Loader)
+        self.operator = self._get_operator()
+        self.yaml_file_name = self.path.stem
+
+    def _get_operator(self) -> str:
+        operator = list(self.yaml_content.keys())[0]
+        if operator not in YamlFile.operator_instance_builder_callable_map:
+            raise NotImplementedError(f"Operator support for {operator} not available")
+        return operator
+
+    def to_operator(self) -> LoadFileOperator:
+        return self.operator_instance_builder_callable_map[self.operator](
+            self.yaml_content[self.operator], self.yaml_file_name
+        )
+
+
+def get_sql_files(directory: Path, target_directory: Path | None) -> set[SqlFile]:
+    """
+    Get all sql files within a directory.
+
+    :param directory: The directory look in for sql files.
+    :param target_directory: The target directory path for the executable sql.
+
+    :returns: the workflow files found in the directory.
+    """
+    return {
+        SqlFile(root_directory=directory, path=child, target_directory=target_directory)  # type: ignore
+        for child in directory.rglob(f"*.{SqlFile.file_type}")
+        if child.is_file() and not child.is_symlink()
+    }
+
+
+def get_yaml_files(directory: Path, target_directory: Path | None) -> set[YamlFile]:
+    """
+    Get all sql files within a directory.
+
+    :param directory: The directory look in for yaml files.
+    :param target_directory: The target directory path for the executable yaml workflow.
+
+    :returns: the workflow files found in the directory.
+    """
+    return {
+        YamlFile(root_directory=directory, path=child, target_directory=target_directory)  # type: ignore
+        for child in directory.rglob(f"*.{YamlFile.file_type}")
+        if child.is_file() and not child.is_symlink()
+    }
```

Comment:
These two methods are super similar. Would it make sense to unify them into a single `get_files_by_type`?